### ISSUE TITLE
Very small training loss values

### ISSUE THREAD

**Issue by T02097** [explainingai-code/DDPM-Pytorch#10]
Hello, I got a loss value of just 0.0218 for the first epoch in the second training, because I didn't pay attention to the loss value in the first training, and I would like to ask if such a small loss value is the right thing to do.

**Comment by explainingai-code** [comment]
Hello @JiangHaiting ,
Can you clarify a bit on what you mean by first training and second training ?
Also are you training on mnist dataset ?

**Comment by T02097** [comment]
I used your code for the first training a month and a half ago, when it was configured for the MNIST dataset, and ran the sampling file as well at that time, and I left it alone after I got the correct results, but now I want to draw a loss curve graph, so I did a second training on this code, but this time the training started out with a very low loss value of only 0.0218
&nbsp;


宀
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月16日(星期一) 下午2:22
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [explainingai-code/DDPM-Pytorch] Very small training loss values (Issue #10)














 
Hello @JiangHaiting ,
 Can you clarify a bit on what you mean by first training and second training ?
 Also are you training on mnist dataset ?

**Comment by explainingai-code** [comment]
I am guessing the low loss value is because training resumed from the checkpoint that you last saved. 
Can you comment this block [here](https://github.com/explainingai-code/DDPM-Pytorch/blob/main/tools/train_ddpm.py#L49-L52) or rename your saved checkpoint to a different name and then run training again. 
Once you do that you should see a value of 0.2-0.3 after first epoch and then after entire training is finished, you should see values like 0.02.

**Comment by T02097** [comment]
Thank you very much! But since I'm a newbie, I'd like to ask if it's better to rename the ddpm_ckpt.pth file that I got from the previous training completion?
Also, when I look at the README file you gave me again, I see in the Output that it says “creates a folder with the task_name key in the configuration file every time you run it”, but I don't see this folder when I run it, so do I need to create it manually?

&nbsp;


宀
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月16日(星期一) 下午2:33
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [explainingai-code/DDPM-Pytorch] Very small training loss values (Issue #10)














 
I am guessing the low loss value is because training resumed from the checkpoint that you last saved.
 Can you comment this block here or rename your saved checkpoint to a different name and the run training again.
 Once you do that you should see a value of 0.2-0.3 after first epoch and then after entire training is finished, you should see values like 0.02.

**Comment by explainingai-code** [comment]
Yeah renaming works, as it allows you to pause and resume the current training run that you are interested in.
So just rename the old checkpoint and then ddpm_ckpt.pth will be the checkpoint saved for your new run. 

Regarding the folder, it should be created every time you run - https://github.com/explainingai-code/DDPM-Pytorch/blob/main/tools/train_ddpm.py#L45 (if it already does not exist).
In your `DDPM-Pytorch` folder, is there no sub-folder named `default` or whatever is the value of this [key](https://github.com/explainingai-code/DDPM-Pytorch/blob/main/config/default.yaml#L22) (in case you changed it)

**Comment by T02097** [comment]
In my DDPM-Pytorch folder there is a subfolder named default, my latest model checkpoints are stored directly under default, as long as I can save the latest checkpoints, will it matter if I don't have a task_name subfolder?
&nbsp;


宀
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月16日(星期一) 下午2:45
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [explainingai-code/DDPM-Pytorch] Very small training loss values (Issue #10)














 
Yeah renaming works, as it allows you to pause and resume the current training run that you are interested in.
 So just rename the old checkpoint and then ddpm_ckpt.pth will be the checkpoint saved for your new run.
 
Regarding the folder, it should be created every time you run - https://github.com/explainingai-code/DDPM-Pytorch/blob/main/tools/train_ddpm.py#L45 (if it already does not exist).
 In your DDPM-Pytorch folder, is there no sub-folder named default or whatever is the value of this key (in case you changed it)

**Comment by explainingai-code** [comment]
In the README when I say task_name folder , I mean the value for the key `task_name` in config file(default.yaml), which is [`default`](https://github.com/explainingai-code/DDPM-Pytorch/blob/main/config/default.yaml#L22).
So as long as you have a `default` folder, everything is fine.
Sorry about the confusion.

**Comment by T02097** [comment]
Ok ok, again thank you so much for answering my questions, have a great life!
&nbsp;


宀
***@***.***







&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年12月16日(星期一) 下午2:59
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [explainingai-code/DDPM-Pytorch] Very small training loss values (Issue #10)














 
In the README when I say task_name folder , I mean the value for the key task_name in config file(default.yaml), which is default.
 So as long as you have a default folder, everything is fine.
 Sorry about the confusion.

**Comment by explainingai-code** [comment]
No problem at all :) Feel free to re-open the issue and ask, incase you end up having any doubts regarding this.
